### ISSUE TITLE
Should Laser Tanks require power?

### ISSUE THREAD

**Issue by ZekeDlyoung** [TheSuperHackers/GeneralsGamePatch#859]
In VZH Laser Tanks require 1 power to operate, this is not only a weird mechanic, but it creates a few issues:

1. It makes Laser Gen pretty defenseless when low powered
2. Even while you are low powered, Laser Tanks will still continue to build and clog up your War Factory for a while even after your power returns
3. It requires a larger investment to build an army with Laser Gen as you have to build power plants as well, which seems like a lot for a unit that's only marginally better than a Crusader

### Proposal 1

Remove power requirement entirely

### Proposal 2

Give it a global control rods upgrade that cancels out its power consumption, so it still gets disabled when you get low powered, but it doesn't slowly eat up your power when spammed

### Proposal 3

Improve its performance to make it worth having to go through all the trouble of building power plants (i.e. give it explosion damage to make it good against infantry as well)

### Proposal 4

Keep Laser Tank Power consumption, but do not power them off on low power.

**Comment by xezon** [comment]
I expect heated discussion :p

**Comment by ZekeDlyoung** [comment]
> I expect heated discussion :p

Me too 😛

**Comment by commy2** [comment]
> It makes Laser Gen pretty defenseless when low powered

That is the whole point. That's pretty much the entire lore of Lasergen xD

**Comment by ZekeDlyoung** [comment]
> > It makes Laser Gen pretty defenseless when low powered
> 
> That is the whole point. That's pretty much the entire lore of Lasergen xD

Indeed. The bigger issue for me is the "require 1 power". I can live with them being disabled, but I hope the requirement can be canceled out at least 🙃

**Comment by xezon** [comment]
Laser Tanks typically come in lower numbers in matches. It is a risk to build them due the risk of power outage, therefore it is almost always better to just spam Humvees instead.

**Comment by Stubbjax** [comment]
It is a weird, counterintuitive and often frustrating mechanic that lowers strategic diversity and tends to catch a lot of players off-guard. I don't see many upsides to the behaviour. It felt much nicer to play when ShockWave removed it.

It would certainly help in common early-game situations where laser crusaders are fending off sneaky gattling tanks and a GLA bombs the power, effectively ending Townes' ~carrier~ career.

**Comment by xezon** [comment]
Perhaps they could have a different penalty for low power?

**Comment by ZekeDlyoung** [comment]
> Perhaps they could have a different penalty for low power?

Not possible afaik

**Comment by ImTimK** [comment]
in 1.06 they increased power requirement still, but didn't power off on low power.

Idk Laser Tanks aren't that special, they're obviously very strong but not sure if low power should be that punishing, especially if he spammed more of them. Direct competitors don't have this downside.

**Comment by ImTimK** [comment]
> Perhaps they could have a different penalty for low power?

Would be cool if turret can still operate, but can't move.

**Comment by penfriendz** [comment]
I am in favour of simply removing the power mechanic. While it will change gameplay, and I don't hate the mechanic, it is a little strange and will simplify the game. From a balance perspective laser tanks aren't terribly overpowered or overused.

**Comment by ImTimK** [comment]
More Laser Tanks means less Vees :)

**Comment by MTKing4** [comment]
Laser tank Spam owns in late game big economy games though, in FFAs especially, combo-ed with avengers, they crack every other army like biscuits, even Air force, and power is no big issue there.

**Comment by ImTimK** [comment]
> Laser tank Spam owns in late game big economy games though, in FFAs especially, combo-ed with avengers, they crack every other army like biscuits, even Air force, and power is no big issue there.

Yes, however USA can do the same, albeit more expensive, but at this point this doesn't matter as much anymore in huge eco games.

**Comment by MTKing4** [comment]
Laser is better because it's not just about Laser tanks, you also have the best defense in the game, laser Patriots

**Comment by ImTimK** [comment]
> Laser is better because it's not just about Laser tanks, you also have the best defense in the game, laser Patriots

Overall laser is better in ffa for sure. Offensively the only problem he has are nuke cannons and infernos, which tomahawks can take care of easy.

**Comment by commy2** [comment]
Laser Power Plant produces 8 power (16 with upgrade). Normal and AFG USA Power Plants are 5/10 respectively.

It would feel weird if Lasergen cared less about power than other USAs.

**Comment by xezon** [comment]
It is nice though that Laser Powerplant already compensates a bit for the Laser Tank power requirement. This makes it appear less of a burden.

**Comment by Stubbjax** [comment]
> Laser Power Plant produces 8 power (16 with upgrade). Normal and AFG USA Power Plants are 5/10 respectively.
> 
> It would feel weird if Lasergen cared less about power than other USAs.

It should also be noted that Laser Defense Turrets consume 5 power (despite what the tooltip says) vs 3 for Patriot Batteries.

**Comment by ImTimK** [comment]
> It is nice though that Laser Powerplant already compensates a bit for the Laser Tank power requirement. This makes it appear less of a burden. If Laser Tank power consumption was removed, Power Plant likely would need tweaking too.

In the 1.06 example I mentioned above they made it so that laser tanks still add to the power requirement, but don't shut off when power is low.

**Comment by xezon** [comment]
### Proposal 4

Keep Laser Tank Power consumption, but do not power them off on low power.

**Comment by ZekeDlyoung** [comment]
> It is nice though that Laser Powerplant already compensates a bit for the Laser Tank power requirement. This makes it appear less of a burden. If Laser Tank power consumption was removed, Power Plant likely would need tweaking too.

Speaking of which are there plans of using Laser Gen's unused purple control rods for this patch to visually show that he has better reactors?
https://www.moddb.com/mods/nproject-mod/images/advanced-cold-fusion-reactor#imagebox

**Comment by commy2** [comment]
> It should also be noted that Laser Defense Turrets consume 5 power (despite what the tooltip says) vs 3 for Patriot Batteries.

[Fixed Russian](https://github.com/TheSuperHackers/GeneralsGamePatch/pull/860/files). The others seem already to be correct.

**Comment by xezon** [comment]
> Speaking of which are there plans of using Laser Gen's unused purple control rods for this patch to visually show that he has better reactors?

Can do. Superweapon General also has different color.

**Comment by commy2** [comment]
SWG produces 5 only. The difference is the Control Rods upgrade. So any visual difference should exist before Control Rods.

**Comment by ImTimK** [comment]
> > Speaking of which are there plans of using Laser Gen's unused purple control rods for this patch to visually show that he has better reactors?
> 
> 
> 
> Can do. Superweapon General also has different color.

Red would be cool, best associated with Laser I think.

**Comment by xezon** [comment]
* #861

**Comment by ReLaX82** [comment]
I am totally against the removal as imo it adds game depth and contributes to different, unique play styles vs Laser for example as Chinav, Tank or Nuke.

I agree with less Vees overall is a good approach.

For me there are a few propsals thinkable.

1 Reduce Power consumption of Laser Tanks (mabye to 50% only)

2 Reduce Cannon Fire rate and/or speed if Laser gen in low Power condition as its not too drastic this way then powering off completly.

**Comment by ZekeDlyoung** [comment]
> Reduce Power consumption of Laser Tanks (mabye to 50% only)

Laser tank power consumption is already set to 1, and decimal values can't be used, so it can't be lowered further

> Reduce Cannon Fire rate and/or speed if Laser gen in low Power condition as its not too drastic this way then powering off completly.

Not possible. Though I once had a similar idea where Laser Tanks would have Gattling Tank rapid fire, but backwards to they fire slower over time to simulate overheating

**Comment by ReLaX82** [comment]
Is it possible that it still shoots and not moves OR still moves and not shoots?

**Comment by ImTimK** [comment]
> Is it possible that it still shoots and not moves OR still moves and not shoots?

Actually non functional turret would make the most sense then, I like this idea.

**Comment by ZekeDlyoung** [comment]
Afaik, it's not possible to invoke any other low power penalty on an object other than full shut down, as the system is closed. Objects either use power or they don't, it doesn't have other fields that can be modified, and there is no "low powered state" that can be used as a trigger for other systems

**Comment by ReLaX82** [comment]
What about an upgrade that you can get somewhere to be free from Power consumption?

**Comment by ZekeDlyoung** [comment]
> What about an upgrade that you can get somewhere to be free from Power consumption?

That's proposal 2 more or less

**Comment by penfriendz** [comment]
I don't like adding a new upgrades for this. It is kinda feel changing in a more obvious way than balance. Making it not use any power but disable when on low power feels good to me.

**Comment by xezon** [comment]
From my understanding this Upgrade is nothing that player sees.

**Comment by penfriendz** [comment]
I wanted to be clear about the effect. In that case I agree

**Comment by FDPUser443** [comment]
Proposal 5 ?
Laser Tanks Power requirement = Max(LTC, 7);
LTC - Laser Tanks count

**Comment by xezon** [comment]
I doubt cap is possible. Each unit acts on its own and consumes whatever it consumes.

**Comment by RisingZH** [comment]
I don’t like this, what problem are we trying to solve with this change? Seems unnecessary.

**Comment by Stubbjax** [comment]
Does the weakness not also seem unnecessary? I think Zeke summed it up pretty well. The fact that it disincentivises tank usage over Humvees is also important to consider.

**Comment by RisingZH** [comment]
I tried to start a discussion a while ago and it didn’t really get off the ground, but aren’t we still trying to keep the new patch feel like 1.04 as possible and fix the biggest imbalances?

This change would make the new patch feel less like 1.04. If you play the generals challenge Lasers whole weakness is meant to be low power. And it doesn’t fix any imbalance, nobody is playing 1.04 thinking laser sucks because my tanks need power.

Just think if we go down the road of changing stuff like this we’ll end up with patch notes 50 pages long and a lot of people will be put off because it’s not the game they are used to.

**Comment by Stubbjax** [comment]
Very good points. The change would be fairly visible and the campaign would be negatively impacted. I think this outweighs any benefits in this case.

Regarding the patch notes, they ironically have to be longer for any respective balance tweaks to be less visible. Players are less likely to notice 100 subtle balance tweaks than 10 major ones.

**Comment by commy2** [comment]
Laser tank does 33% more damage per shell than most other tanks (Crusader/Paladin/(Nuke) Battle Master/Marauder). 80 vs 60.

**Comment by xezon** [comment]
Laser General does not have Tomahawk in its arsenal, so it is nice to have stronger tanks at least.

**Comment by ReLaX82** [comment]
Thinking this over again makes me think we dont need this change at all.

**Comment by xezon** [comment]
Proposal 3 is not a bad idea on first sight. Then it performs more like the Nuclear Battlemaster, which is a special variant with Radiation Shells - better against infantry units.

**Comment by xezon** [comment]
| Object  | Time to kill 1 Missile Defender |
|-|-|
| USA Crusader Tank | 00:32 |
| USA Laser Tank | 00:24 |
| USA Laser Tank + Hellfire Drone | 00:04 |
| Battlemaster | 00:32 |
| Nuke Battlemaster | 00:03 |
| GLA Scorpion + Rocket | 00:01 |
| GLA Scorpion + Anthrax Beta Shells | 00:03 |
| GLA Marauder | 00:32 |

**Comment by MTKing4** [comment]
I'm not sure this is a good idea, laser tank is already the one of best tanks in the game

**Comment by xezon** [comment]
Close?

**Comment by MTKing4** [comment]
Yeah you can close

**Comment by ImTimK** [comment]
I think low power penalty currently is very very harsh. One Saboteur can render an entire army useless, when easily can result in a loss. Tank spam as Vanilla USA might less (cost) effective but isn't completely screwed when this happens.

Proposal: Lowering penalty by disabling the turret but keeping ability to drive.

Or reverse: Keeping it immobilized but with operable turret.

**Comment by xezon** [comment]
Reopen or keep close?

**Comment by xezon** [comment]
The way I see it that Laser Tanks are a liability. It is better to not build them. 2 or so enemy Auroras are enough to shutdown Laser tanks by targeting upgraded Power plants.

**Comment by ImTimK** [comment]
Keep open I think.

**Comment by MTKing4** [comment]
> Proposal: Lower penalty by disabling the turret but keeping the ability to drive.

If we can agree on this then yeah sure it's a good change that i favor, otherwise no point keeping it open if we can't agree on anything.

**Comment by xezon** [comment]
> Proposal: Lower penalty by disabling the turret but keeping the ability to drive.

Reads good to me. Allows affected player to retreat in case of emergency and therefore lowers the penalty. This would require code changes though I think. A feature to power off the weapon only.

**Comment by xezon** [comment]
> Improve its performance to make it worth having to go through all the trouble of building power plants (i.e. give it explosion damage to make it good against infantry as well)

This is an interesting option too. It looks odd that the Crusader Laser is so weak vs infantry, when the Laser Turret is so much stronger. If performance of Laser Tank is improved, then it can be a real option to use more often early and risk its penalties. It would also combine well with reduced effectiveness of Pathfinders (#690).

**Comment by commy2** [comment]
Laser tank already does 80 damage vs 60 of Crusader/Paladin/Battlemaster/Marauder. That's +33%, a very good upgrade.

**Comment by MTKing4** [comment]
I really feel that laser tanks are are already too good and not in need of any kind of buff

**Comment by xezon** [comment]
> I really feel that laser tanks are are already too good and not in need of any kind of buff

Hmm. But if this was true, why are they so rarely build? In World Series matches you never see them. Humvees all the way.

**Comment by commy2** [comment]
That's because Humvees are OP thanks to S&D MD range, Ambo repair and cheap Scout Drones with 3.5 times as many hp as in CCG. You're not going to change that by buffing the Laser tank without screwing over Laser vs Tank/Nuke/vChina tank battle balance.

**Comment by MTKing4** [comment]
Like commy said, and also because they're slower than Vees.

In Laser vs chinas is where you see them built, they're very strong against things like overlords and emperors, they also demolish every other army with their laser tank + Avenger combo in lategame FFAs

**Comment by xezon** [comment]
That is as long as the power is up.

**Comment by MTKing4** [comment]
Usually in those games laser would've built a million powerplant surrounded with laser turrets everywhere, you can try to stop it by killing the power, but it's almost pointless at this point as you need to kill way to many powerplants before he's out of power, and even then, he can rebuild very quickly

**Comment by ReLaX82** [comment]
Those tanks are mostly used in team games and 1v1 vs Nuke, vChina or Tank.
The option to have it more useful vs infantry I like as well.
Go for the option to disable the laser while on low power maybe and increase effectiveness vs infantry.

**Comment by commy2** [comment]
I tried if POWERED state can be combined with portable turrets like Overlord and Avenger, to make it so only the turret can be shut down.

But adding POWERED to the Avenger turret crashes the game once you lose power, so that doesn't work.

**Comment by xezon** [comment]
Of course it crashes :D

**Comment by MTKing4** [comment]
Close until thyme then?

**Comment by fastAiRR** [comment]
I just randomly saw this post, so thought I would leave my opinion, even if outdated.

Its stupid that the laser tank turns off when it loses power, after all, its not exactly got a cable plugged in the back of it, running from your base. Surely it has a battery or even, it generates its own power. 

But, killing the power plants of an enemy to disable his massive laser tank army is such a cool and unique strategy. It should therefore never be removed from the game. If it was, you are killing the character of the game, one of the very things that makes it what it is.

**Comment by xezon** [comment]
mrCorat via #2522:

> Laser tanks - unique unit for laser general. And in game realities it should give unique gameplay for this general. The nearest in gameplay style is vanilla USA. But there is some properties that makes gameplay in vanilla more preferable, then laser general:

> * Laser tanks requires energy (harder to play by tank spam)
> * No artillery (harder to break huge defense and the longest strike distance have rocket defenders).

> And by first property we can't spam much laser tanks, so in any case opponent spam, especially in tanks, better then for laser general. But it core gameplay (included in general physics).
That's why I propose to increase strike distance for laser tanks (more then rocket defenders, less than artillery). I think distance of rocket buggy more real.
